### DISCUSSION TITLE
Will the Expand API ever support contextual tuples?

### DISCUSSION THREAD

**Discussion by laerva1801** [openfga/community#375] [Q&A] [answered]
Apologies if this has been answered before elsewhere, I could not find any resources on this.

The Expand API does **not** support contextual tuples. This is also stated [in the docs](https://openfga.dev/docs/modeling/contextual-time-based-authorization#summary). My question is whether this is a temporary limitation of OpenFGA at this stage of development, or if this is an intentional decision that will not change.

I'm implementing OpenFGA in a context where significant parts of the authorization facts come from contextual tuples. This works great for the Check API and ListObjects, but limits my options for explaining how access is obtained via the Expand API.

Cheers.

**Comment by aaguiarz** [comment, marked as answer]
Hi!

There's an open issue https://github.com/openfga/openfga/issues/951, and an open PR that still needs work https://github.com/openfga/api/pull/202
